api/src/main/java/io/grpc/NameResolverProvider.java

```diff
@@ -62,4 +65,14 @@ public abstract class NameResolverProvider extends NameResolver.Factory {
   protected String getScheme() {
     return getDefaultScheme();
   }
+
+  /**
+   * Returns the {@link SocketAddress} types this provider's name-resolver is capable of providing.
+   * This enables selection of the appropriate {@link ManagedChannelProvider} for a channel.
+   *
+   * @return the {@link SocketAddress} types this provider's name-resolver is capable of providing.
+   */
+  protected Collection<Class<? extends SocketAddress>> getSupportedSocketAddressTypes() {
```

Comment:
I don't really understand how "supported" works on NR. I understand it for ManagedChannelBuilderProvider. Maybe getSocketAddressTypes, getReturnedSocketAddressTypes, getProducedSocketAddressTypes?